### PR TITLE
check values file exist before package install & update

### LINKED ISSUE #800: The "tanzu package install" command should validate values file exists before doing anything.
## Bug Report
The tanzu package install command does not check that the values file exists before doing anything else.

```
    $ tanzu package install contour --package-name contour.community.tanzu.vmware.com --version 1.18.1 --values-file contour-config.yaml
    / Installing package 'contour.community.tanzu.vmware.com'
    | Getting namespace 'default'
    | Getting package metadata for 'contour.community.tanzu.vmware.com'
    | Creating service account 'contour-default-sa'
    | Creating cluster admin role 'contour-default-cluster-role'
    | Creating cluster role binding 'contour-default-cluster-rolebinding'

    Please consider using 'tanzu package installed delete' to delete the already created associated resources
    | Creating secret 'contour-default-values'

    Error: failed to read from data values file 'contour-config.yaml': open contour-config.yaml: no such file or directory
    Usage:
      tanzu package install INSTALLED_PACKAGE_NAME --package-name PACKAGE_NAME --version VERSION [flags]
````

The result is that it creates service accounts, roles and role bindings before failing.

If you correct the problem with values file name and try again, it then fails since those resources already exist.

```
    $ tanzu package install contour --package-name contour.community.tanzu.vmware.com --version 1.18.1 --values-file ~/Desktop/contour-config.yaml
    / Installing package 'contour.community.tanzu.vmware.com'
    | Getting namespace 'default'
    | Getting package metadata for 'contour.community.tanzu.vmware.com'
    | Creating service account 'contour-default-sa'

    Error: failed to create ServiceAccount resource: serviceaccounts "contour-default-sa" already exists
    Usage:
      tanzu package install INSTALLED_PACKAGE_NAME --package-name PACKAGE_NAME --version VERSION [flags]
```

It is then necessary to delete the partially installed package before can try again.

```
    $ tanzu package installed delete contour
    Deleting installed package 'contour' in namespace 'default'. Are you sure? [y/N]: y
    | Uninstalling package 'contour' from namespace 'default'
    / Getting package install for 'contour'

    package 'contour' is not installed in namespace 'default'. Deleted previously installed resources
```

The last message there is a bit confusing in that it says the package was not installed, yet is still deleting stuff.

## Expected Behavior
Should check that the values file exists before doing anything.

## Steps to Reproduce the Bug
Run:
```
tanzu package install contour --package-name contour.community.tanzu.vmware.com --version 1.18.1 --values-file non-existant-file.yaml
```

##Screenshots or additional information and context
Nothing.

##Environment Details
Build version (tanzu version):
version: v0.2.1
buildDate: 2021-09-29
sha: ceaa474

##Diagnostics and log bundle
Nothing.

**Original issue in Tanzu Community Edition:** https://github.com/vmware-tanzu/community-edition/issues/2185

### PR DESCRIPTION
**Pull request by ggpaue**
### What this PR does / why we need it
check values file exist before package install & update command executes
### Which issue(s) this PR fixes

Fixes #
https://github.com/vmware-tanzu/tanzu-framework/issues/800
### Describe testing done for PR
manually test on cluster

### Release note

```release-note
check values file exist before package install & update command executes
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/tkgpackageclient/package_update.go [hunk 1]
```diff
@@ -38,6 +38,13 @@ func (p *pkgClient) UpdatePackage(o *tkgpackagedatamodel.PackageOptions, progres
 		}
 	}()
 
+	if o.ValuesFile != "" {
+		if _, err = ioutil.ReadFile(o.ValuesFile); err != nil {
+			err = errors.Wrap(err, fmt.Sprintf("failed to read from data values file '%s'", o.ValuesFile))
+			return
+		}
+	}
+
```
Comment (maralavi) [inline pkg/v1/tkg/tkgpackageclient/package_update.go:47]: This check should have been put after the following block, as we do not want to check twice for the isntall case

```
if pkgInstall == nil {
		if !o.Install {
			err = &tkgpackagedatamodel.PackagePluginNonCriticalError{Reason: tkgpackagedatamodel.ErrPackageNotInstalled}
			return
		}
		progress.ProgressMsg <- fmt.Sprintf("Installing package '%s'", o.PkgInstallName)
		p.InstallPackage(o, progress, tkgpackagedatamodel.OperationTypeUpdate)
		return
	}
```

File: pkg/v1/tkg/tkgpackageclient/package_install.go [hunk 2]
```diff
@@ -62,6 +62,13 @@ func (p *pkgClient) InstallPackage(o *tkgpackagedatamodel.PackageOptions, progre
 		return
 	}
 
+	if o.ValuesFile != "" {
+		if _, err = ioutil.ReadFile(o.ValuesFile); err != nil {
+			err = errors.Wrap(err, fmt.Sprintf("failed to read from data values file '%s'", o.ValuesFile))
+			return
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (ggpaue) [reply pkg/v1/tkg/tkgpackageclient/package_install.go:68]: thanks for sharing the change @maralavi I have tested the change and it worked

File: pkg/v1/tkg/tkgpackageclient/package_update.go (hunk 1 above)
Comment (ggpaue) [reply pkg/v1/tkg/tkgpackageclient/package_update.go]: changes updated

**Review by maralavi — APPROVED** [review]

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Review by danniel1205 — APPROVED** [review]

**Pull request merged**